src/common/arrayUtils.ts

```diff
@@ -0,0 +1,11 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+export function asArray<T>(thing: T | ReadonlyArray<T>): T[] {
```

Comment:
Ideally this would return a `ReadonlyArray<T>` when given a `ReadonlyArray<T>`, and return a `T[]` when given a `T[]`. That consumed too much time, I suppose I'm missing a TS trick.